analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -222,4 +222,677 @@ public class Test {
+    [DataRow("__refvalue(tf, [t:Exception])", false)]
+    [DataRow("__refvalue(tf, [u:System].Exception)", true)]
+    [DataRow("__refvalue(tf, System.[t:Exception])", false)]
+    public void IdentifierToken_Type_RefValue(string refValueExpression, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes(
+            $$"""
+              using System;
+
+              public class Test
+              {
+                  public void M(TypedReference tf)
+                  {
+                      Exception tfValue = {{refValueExpression}};
+                  }
+              }
+              """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("default([t:Exception])", false)]
+    [DataRow("default([u:System].Exception)", true)]
+    public void IdentifierToken_Type_DefaultValue(string defaultValueExpression, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes(
+            $$"""
+              using System;
+
+              public class Test
+              {
+                  public void M()
+                  {
+                      var x = {{defaultValueExpression}};
+                  }
+              }
+              """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("sizeof([t:Int32])", false)]
+    [DataRow("sizeof([u:System].Int32)", true)]
+    public void IdentifierToken_Type_SizeOfValue(string sizeOfExpression, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes(
+            $$"""
+              using System;
+
+              public class Test
+              {
+                  public void M()
+                  {
+                      var x = {{sizeOfExpression}};
+                  }
+              }
+              """ /*, allowSemanticModel */);
+
+#if NET
+
+    [DataTestMethod]
+    [DataRow("stackalloc [t:Int32][2]", false)]
+    [DataRow("stackalloc [u:System].Int32[2]", true)]
+    [DataRow("stackalloc [u:Int32]", true, true)] // compilation error. Type can not be resolved (must be an array type)
+    public void IdentifierToken_Type_StackAlloc(string stackAllocExpression, bool allowSemanticModel = true, bool ignoreCompilationErrors = false) =>
+        ClassifierTestHarness.AssertTokenTypes(
+            $$"""
+              using System;
+
+              public class Test
+              {
+                  public void M()
+                  {
+                      Span<int> x = {{stackAllocExpression}};
+                  }
+              }
+              """, true, ignoreCompilationErrors);
+
+#endif
+
+    [DataTestMethod]
+    [DataRow("""
+        from [t:Int32] x in new long[10]
+        select x
+        """, false)]
+    [DataRow("""
+        from [u:System].Int32 x in new long[10]
+        select x
+        """, true)]
+    [DataRow("""
+        from x in new long[10]
+        join [t:Int32] y in new long[0] on x equals y into g
+        select g
+        """, false)]
```

Comment:
I think it would be easier to read if the cases would be single line.

Reply:
We used the same formatting in IdentifierToken_BaseTypeList before. The given syntax is the standard convention for QueryComprehensions. I would like to keep it this way.